### DISCUSSION TITLE
Particle Ordering By Scale

### DISCUSSION THREAD

**Discussion by skyelynwaddell** [godotengine/godot-proposals#10285] [Engine Core]
### Describe the project you are working on

Often times when creating Particle systems, especially in the background Parallax effect could be easily achieved if there was an option to sort Ordering by Particle Size.
This way it could properly simulate for example , cloud that are further than closer.

### Describe the problem or limitation you are having in your project

Im finding some of my clouds that are bigger draw over the smaller clouds, and same with other background  elements.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It adds a very simple touch of polish for the particle system that i feel would be used very often.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add a checkbox in ParticleProccessMaterial called Sort By Scale

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think it would be a useful option that would see a lot of use.

### Is there a reason why this should be core and not an add-on in the asset library?

It seems like it should have been in the Particle system already due to the way Time works.
```[tasklist]
### Tasks
```

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.
